src/Controllers/AuthController.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+namespace Railroad\MusoraApi\Controllers;
+
+use Exception;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Request;
+use Illuminate\Routing\Controller;
+use Railroad\Ecommerce\Events\AppSignupStartedEvent;
+use Railroad\MusoraApi\Contracts\UserProviderInterface;
+use Railroad\MusoraApi\Requests\CreateIntercomUserRequest;
+
+class AuthController extends Controller
+{
+    /**
+     * @var UserProviderInterface
+     */
+    private $userProvider;
+
+    /**
+     * AuthController constructor.
+     *
+     * @param UserProviderInterface $userProvider
+     */
+    public function __construct(
+        UserProviderInterface $userProvider
+    ) {
+        $this->userProvider = $userProvider;
+    }
+
+    /**
+     * @param Request $request
+     * @return JsonResponse
+     */
+    public function updateUser(Request $request)
+    {
+        if ($request->has('file')) {
+            $this->userProvider->setCurrentUserProfilePictureUrl($request->get('file'));
+        }
+
+        if ($request->has('phone_number')) {
+            $this->userProvider->setCurrentUserPhoneNumber($request->get('phone_number'));
+        }
+
+        if ($request->has('display_name')) {
+            $updatedUser = $this->userProvider->setCurrentUserDisplayName($request->get('display_name'));
+            if (!$updatedUser) {
```

Comment:
Controller methods multiple return statements, and a few if statements, may be replaced with custom exceptions. This would increase code readability and decreases complexity.
Example of custom exception: https://github.com/railroadmedia/railcontent/blob/1.0-/src/Exceptions/NotAllowedException.php